pkg/asset/agent/installconfig.go

```diff
@@ -255,6 +255,31 @@ func (a *OptionalInstallConfig) GetBaremetalHosts() []*baremetal.Host {
 	return nil
 }
 
+// ValidateBMCConfiguration checks that the Provisioning fields are configured when baremetal hosts BMC is set.
+func (a *OptionalInstallConfig) ValidateBMCConfiguration() {
```

Comment:
This method seems to be used only in `agenthosts.go`, so it could be moved directly there (and made it private, if not needed by anyone else)

Reply:
Yes, its only used once but it has to access the `provisioning` fields in install-config, beyond just the host fields, so it seems more logical to keep it there. It also keeps all of the install-config related warnings together.

Reply:
Right, but I think there's a potential issue in its usage, please see the other comment where `installConfig.ValidateBMCConfiguration()` it's used